test/common/IotHub.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace common
+{
+    using System;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Common;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Azure.EventHubs;
+    using Newtonsoft.Json;
+
+    using DeviceTransportType = Microsoft.Azure.Devices.TransportType;
+    using EventHubTransportType = Microsoft.Azure.EventHubs.TransportType;
+
+    public class IotHub
+    {
+        readonly string eventHubEndpoint;
+        readonly string iotHubConnectionString;
+        readonly Lazy<RegistryManager> registryManager;
+        readonly Lazy<ServiceClient> serviceClient;
+        readonly Lazy<EventHubClient> eventHubClient;
+
+        public string Hostname =>
+            IotHubConnectionStringBuilder.Create(this.iotHubConnectionString).HostName;
+        public string EntityPath =>
+            (new EventHubsConnectionStringBuilder(this.eventHubEndpoint)).EntityPath;
+
+        RegistryManager RegistryManager => this.registryManager.Value;
+        ServiceClient ServiceClient => this.serviceClient.Value;
+        EventHubClient EventHubClient => this.eventHubClient.Value;
+
+        public IotHub(string iotHubConnectionString, string eventHubEndpoint, Option<string> proxyUrl)
```

Comment:
check not null for iot Hub connection string, and event hub endpoint; uses Option<Uri> for proxy url.

Reply:
I'm already checking those in `Program.OnExecuteAsync`. Interally, I assume strings are not null.